### DISCUSSION TITLE
How to use one application (ams console app) that uses a separate mongodb.

### DISCUSSION THREAD

**Discussion by rahul78275** [ant-media/Ant-Media-Server#5445] [Plugins/Custom Extensions]
Hello Team

I am developing an application on top of antmedia server, where I am creating some rest apis and want to use a separate **mongodb** for my application. Other default apps (LiveApp, webRTCAppEE) remains unchanged.

What/Where I have to make changes in files ?

Also in DB connection credentials do i need to pass the instance public IP ?

`store = new CRUDMongoStore("INSTANCE_PUBLIC_IP", "", "", "CRUDstore");`

Would any changes is required if my mongodb is installed on the same instance where ams is installed OR both are on different instance on AWS.

Thank You

**Comment by rahul78275** [comment]
@burak-58 abi 📢

**Reply by mekya** [reply]
Thank you for the gentle reminder @rahul78275. I've also made the gentle reminder ;)

**Comment by USAMAWIZARD** [comment]
Hi @rahul78275  let me check this and get back to you.

**Reply by USAMAWIZARD** [reply]
@rahul78275 I have talked to bruak abi for your issues and he will reply to you soon

**Comment by burak-58** [comment]
Hi my dear friend Rahul,

You know that AMS uses MongoDB in cluster mode. But there is no problem with using MongoDB in  standalone mode as well. 
So my suggestion is that:
- UseMongoDB as datastore instead of default one MapDB. To do that, in the application configuration file set db type to mongo and provide credentials.
- In your custom application you can connect and use the same MongoDB with AMS. You can get the MongoDB credentials from the AppSettings Bean. You can reach AppSettings Bean like here:
https://github.com/burak-58/CRUDSampleApp/blob/3e109735f48275d916bd0155afdffce49a6218e7/src/main/java/io/antmedia/plugin/CRUDMain.java#L21
- Then you can connect the MongoDB with the credentials you got and create your own documents like in this class:
https://github.com/burak-58/CRUDSampleApp/blob/main/src/main/java/io/antmedia/db/CRUDMongoStore.java

**Reply by rahul78275** [reply]
Thank you @burak-58 abi

You are great 🖖. I can recall all configurations now. We had created this plugin together.
Writing api in java just because of you 😄 

Thank you.
